controllers/asosecret_controller.go

```diff
@@ -287,23 +288,33 @@ func (asos *ASOSecretReconciler) createSecretFromClusterIdentity(ctx context.Con
 		return newASOSecret, nil
 	}
 
-	// Fetch identity secret, if it exists
-	key = types.NamespacedName{
-		Namespace: identity.Spec.ClientSecret.Namespace,
-		Name:      identity.Spec.ClientSecret.Name,
-	}
-	identitySecret := &corev1.Secret{}
-	err := asos.Get(ctx, key, identitySecret)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to fetch AzureClusterIdentity secret")
-	}
+	if identity.Spec.CertPath != "" {
```

Comment:
Are you able to add a test case or two to `asosecret_controller_test.go` for this case when CertPath is specified?

Reply:
Good call. I added a test which found a minor error. Service Principal with Certificate was expected to have a client secret [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/9a106f4113b0b81adc7da31641eeb9b00f52bb8e/azure/scope/identity.go#L213-L212). Client Secret isn't needed with using Certificate ([see this in the Azure SDK](https://github.com/Azure/azure-sdk-for-go/blob/0387c89ceebae256c9ea717533c739b7a875d996/sdk/azidentity/environment_credential.go#L115) as an example).